web/src/hooks/queries/useHomePageBlockQuery.ts

```diff
@@ -0,0 +1,138 @@
+import { useQuery } from "@tanstack/react-query";
+
+import { useGraphqlBatcher } from "context/GraphqlBatcher";
+import { useMemo } from "react";
+
+import { graphql } from "src/graphql";
+import { HomePageBlockQuery } from "src/graphql/graphql";
+export type { HomePageBlockQuery };
+
+const homePageBlockQuery = graphql(`
+  query HomePageBlock($blockNumber: Int) {
+    presentCourts: courts(orderBy: id, orderDirection: asc) {
+      id
+      parent {
+        id
+      }
+      name
+      numberDisputes
+      numberVotes
+      feeForJuror
+      stake
+    }
+    pastCourts: courts(orderBy: id, orderDirection: asc, block: { number: $blockNumber }) {
+      id
+      parent {
+        id
+      }
+      name
+      numberDisputes
+      numberVotes
+      feeForJuror
+      stake
+    }
+  }
+`);
+
+export const useHomePageBlockQuery = (blockNumber: number | null, allTime: boolean) => {
+  const isEnabled = blockNumber !== null || allTime;
+  const { graphqlBatcher } = useGraphqlBatcher();
+
+  const usedQuery = useQuery({
+    queryKey: [`homePageBlockQuery${blockNumber}-${allTime}`],
+    enabled: isEnabled,
+    staleTime: Infinity,
+    queryFn: async () => {
+      const data = await graphqlBatcher.fetch({
+        id: crypto.randomUUID(),
+        document: homePageBlockQuery,
+        variables: { blockNumber },
+      });
+      return data;
+    },
+  });
+
+  const courtActivityStats = useMemo(() => {
+    if (usedQuery.data && !usedQuery.isFetching) {
+      // 1. court with most disputes
+      // we only iterate through past courts, since more courts might exist at the present
+      // these diffCourts have: average stakes, and dispute diff
+      const diffCourts = allTime
+        ? usedQuery.data.presentCourts.map((c) => ({
+            ...c,
+            numberDisputes: c.numberDisputes,
+            treeNumberDisputes: c.numberDisputes,
+            numberVotes: c.numberVotes,
+            treeNumberVotes: c.numberVotes,
+            stake: c.stake,
+          }))
+        : usedQuery.data.pastCourts.map((c, i) => ({
+            ...c,
+            numberDisputes: usedQuery.data.presentCourts[i].numberDisputes - c.numberDisputes,
+            treeNumberDisputes: usedQuery.data.presentCourts[i].numberDisputes - c.numberDisputes,
+            numberVotes: usedQuery.data.presentCourts[i].numberVotes - c.numberVotes,
+            treeNumberVotes: usedQuery.data.presentCourts[i].numberVotes - c.numberVotes,
+            stake: (BigInt(usedQuery.data.presentCourts[i].stake) + BigInt(c.stake)) / 2n,
+          }));
```

Comment:
I believe here in the case the courts are new, we want to show the data from zero, i.e. if a new court was created 3 days ago and the user selects "7 Days ago" we should still show the data from the new court from the last 3 days. So you would always iterate through `presentCourts` substracting from `pastCourts` in case it's not `allTime` and there is something at that index.